crypto/crypto/src/hkdf.rs

```diff
@@ -138,14 +141,15 @@ where
 
         for (blocknum, okm_block) in okm.chunks_mut(hmac_output_bytes).enumerate() {
             if let Some(ref prev) = prev {
-                hmac.input(prev)
+                hmac.update(prev)
             }
             if let Some(_info) = info {
-                hmac.input(_info);
+                hmac.update(_info);
             }
-            hmac.input(&[blocknum as u8 + 1]);
+            hmac.update(&[blocknum as u8 + 1]);
 
-            let output = hmac.result_reset().code();
+            let output = hmac.clone().finalize().into_bytes();
+            hmac.reset();
```

Comment:
Instead of cloning and then resetting you should use [`finalize_reset`](https://docs.rs/digest/0.9.0/digest/trait.Digest.html#tymethod.finalize_reset)